### DISCUSSION TITLE
Capitalization on document identifiers

### DISCUSSION THREAD

**Discussion by bobskigit** [ravendb/ravendb#13111] [Q&A]
If we create a document using the studioUI it capitalizes the first letter of the collection name so we get Campaign/id as the document identifier.

However, if we use the c# client it makes the document identifier lower case i.e campaign/id.

Any ideas on how to make it consistent?

**Comment by ayende** [comment]
You can set the document id directly, in which case RavenDB will not modify the case. Note that the studio doesn't actually do anything with the id.

If you save a new document with the following content:

```
{
    "Name": "...",
    "@metadata": {
        "@collection": "Orders"
    }
}
```

The document id will be a Guid. When you _clone_ a document, on the other hand, we'll default to using the collection name as is, like so:
![image](https://user-images.githubusercontent.com/116915/142862593-cab706ec-defd-489a-9d6e-09710975bdd6.png)


I assume that this is what you are talking about? You can just change the casing there.

**Reply by bobskigit** [reply]
How does Raven decide whether to capitalize or not?

We have two collections one called Campaigns and one called PromoandBumperCampaigns, we use the same method to create the objects but for one collection we get campaigns/id and the other we get PromoandBumperCampaigns/id

**Reply by ayende** [reply]
If there is a single word, it will use lower case, if multiple words, it will use as is

**Reply by bobskigit** [reply]
thanks for the quick answer as always.
